aiozk/session.py

```diff
@@ -154,32 +160,37 @@ def __init__(self, servers, timeout, retry_policy, allow_read_only, read_timeout
 
     async def repair_loop(self):
         log.debug('repair loop starting')
-        while not self.closing:
-            log.debug('Await for repairable state')
-            await self.state.wait_for(States.SUSPENDED, States.LOST, loop=self.loop)
-            log.debug('Repair state reached')
-            if self.closing:
-                break
+        try:
+            while not self.closing:
+                log.debug('Await for repairable state')
+                await self.state.wait_for(States.SUSPENDED, States.LOST, loop=self.loop)
+                log.debug('Repair state reached')
+                if self.closing:
+                    break
+
+                await self.find_server(allow_read_only=self.allow_read_only)
+
+                try:
+                    await asyncio.wait_for(self.establish_session(), self.timeout, loop=self.loop)
+                except (exc.SessionLost, asyncio.TimeoutError) as e:
+                    log.info('Session closed: {}'.format(e))
+                    self.conn.abort(exc.SessionLost)
+                    await self.conn.close(self.timeout)  # TODO: make real timeout
+                    self.session_id = None
+                    self.password = b'\x00'
+                    continue
 
-            await self.find_server(allow_read_only=self.allow_read_only)
+                if self.conn.start_read_only:
+                    self.state.transition_to(States.READ_ONLY)
+                else:
+                    self.state.transition_to(States.CONNECTED)
 
-            try:
-                await asyncio.wait_for(self.establish_session(), self.timeout, loop=self.loop)
-            except (exc.SessionLost, asyncio.TimeoutError) as e:
-                log.info('Session closed: {}'.format(e))
-                self.conn.abort(exc.SessionLost)
-                await self.conn.close(self.timeout)  # TODO: make real timeout
-                self.session_id = None
-                self.password = b'\x00'
-                continue
-
-            if self.conn.start_read_only:
-                self.state.transition_to(States.READ_ONLY)
-            else:
-                self.state.transition_to(States.CONNECTED)
-
-            self.conn.start_read_loop()
-            await self.set_existing_watches()
+                self.conn.start_read_loop()
+                await self.set_existing_watches()
+        except Exception as error:
```

Comment:
Which errors do we expect here?

Reply:
If the solely purpose of all this try/catch block is to log error, probably `self.repair_loop_task.add_done_callback` with logging error will look cleaner

Reply:
yep 👍